src/dmd/target.d

```diff
@@ -741,6 +741,64 @@ extern (C++) struct Target
         return global.params.is64bit ? (sz + 7) & ~7 : (sz + 3) & ~3;
     }
 
+    /**
+     * Determine which `in` parameters needs to be passed by `ref`
+     *
+     * Called from `TypeFunction` semantic with the full function type.
+     * This routine must iterate over parameters, and may set `STC.ref_`
+     * for any parameter which already have `STC.in_`.
+     * This hook must never set `STC.ref_` if the parameter is not `STC.in_`,
+     * nor should it ever change anything else.
+     *
+     * This hook will not be called when `-preview=in` wasn't passed to the
+     * frontend, hence it needs not care about `global.params.previewIn`.
+     *
+     * Params:
+     *   tf    = Type of the function to inspect. The type will have its
+     *           parameter types semantically resolved, however other attributes
+     *           (return type, `@safe` / `nothrow`, etc...) must not be used.
+     */
+    extern(C++) void applyInRefParams (TypeFunction tf)
```

Comment:
Needs to be added to target.h, maybe with a cxx-unittest if I was to be stringent about it. :-)

Reply:
Since it's not a virtual function and there's no vtable, I don't see if that'd be useful

Reply:
I need to implement this in gdc, ditto ldc.

Reply:
I added it to `target.h`, just not in `cxx-unittest`